### DISCUSSION TITLE
transparent window blur?

### DISCUSSION THREAD

**Discussion by zydra-zy** [YaLTeR/niri#749] [General]
hi! I'm curious if there are plans for implementing blur on background behind transparent windows?  - or if there are ways of doing this that I'm unaware of!

it's probably the only thing I consider missing for niri to be near perfect for my use and look...

**Comment by YaLTeR** [comment]
It would be good, but it needs some Smithay work first: https://github.com/Smithay/smithay/issues/1363

**Reply by zydra-zy** [reply]
I see!!! I was thinking it was this, but wasn't sure if blur is something that has to be from Smithay or could be done from niri's side regardless of it...
Thanks for the reply! hoping Smithay gets it eventually as it allows for really pretty looking setups..

**Reply by zetaPRIME** [reply]
yeah, my hyprland and i3+picom setups were/are both *heavy* on this; if I was more familiar with how Wayland worked I'd see if I could patch the "surround" into Smithay myself but I honestly barely know the first thing about how the backend works <_<

**Reply by yvvki** [reply]
Any news on the issue?

**Reply by YaLTeR** [reply]
The news is that some (maybe all?) parts of it are doable without that Smithay issue, but nothing implemented yet

**Comment by linuxmobile** [comment]
i'm really waiting for this!

**Comment by visualglitch91** [comment]
Did smithay give up on this? Looks like https://github.com/Smithay/smithay/issues/1363 was deleted :(

**Reply by YaLTeR** [reply]
Idk what happened to the issue but there's a PoC impl in fht-compositor without it so yeah

**Reply by visualglitch91** [reply]
Nice, that looks promissing

**Comment by bezark** [comment]
yeah just chiming in to voice support for this and to get notifications on updates!

**Comment by RandomLegend** [comment]
Hello there - how is the state on this?

I just found out about Niri and swapped from Hyprland and would really love to have the blur^^

**Reply by visualglitch91** [reply]
There's this branch you can compile yourself

https://github.com/YaLTeR/niri/pull/1634

**Reply by RandomLegend** [reply]
Thanks for the info, but i don't like deviating from the main branch on big projects like this.

I'd much rather wait for blur to hit main instead of building some other branch from source...i just find it interesting how blur didn't hit main yet as this is something so fundamental and sought after
